salt/_modules/caasp_hosts.py

```diff
@@ -308,19 +331,32 @@ def get_with_expr(expr):
         raise EtcHostsRuntimeException(
             'Could not add special entries in /etc/hosts: {}'.format(e))
 
+    # sort the names for determinism
+    for ip, names in hosts.items():
+        names.sort()
+
+    # prepend the nodenames at the beginning of each entry
+    try:
+        _add_nodenames_for(hosts, admin_nodes, infra_domain)
+        _add_nodenames_for(hosts, master_nodes, infra_domain)
+        _add_nodenames_for(hosts, worker_nodes, infra_domain)
+        _add_nodenames_for(hosts, other_nodes, infra_domain)
```

Comment:
I would apply the same pattern:

```suggestion
        for x in [admin_nodes, master_nodes, worker_nodes, other_nodes]:
            _add_nodenames_for(hosts, x, infra_domain)
```